pkg/controllers/provisioning/suite_test.go

```diff
@@ -554,19 +557,78 @@ var _ = Describe("Provisioning", func() {
 			Expect(*allocatable.Cpu()).To(Equal(resource.MustParse("4")))
 			Expect(*allocatable.Memory()).To(Equal(resource.MustParse("4Gi")))
 		})
+		It("should schedule based on the max resource requests of containers and initContainers with sidecar containers when initcontainer comes first", func() {
```

Comment:
Ideally, it would be nice to see a few more test cases here around some of the edge-cases that are covered in the `utils/resources/suite_test.go`. I think we're still fine to merge, but I think the thing I feel like we are missing here are cases for multiple native sidecar containers and multiple init containers with different configurations on where they fall in the list.

Both of these tests cases just validate with one each, which makes me feel like we might have gaps if our summation logic is wrong here.